src/structures/Widget.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const Base = require('./Base');
+const Collection = require('../util/Collection');
+
+/**
+ * Represents a Widget.
+ */
+class Widget extends Base {
+  /**
+   * @param {Client} client - The instantiating client
+   * @param {Object} data - The raw data
+   */
+  constructor(client, data) {
+    super(client);
+    this._patch(data);
+  }
+  /**
+   * Builds the widget with the provided data.
+   * @param {*} data The raw data of the widget
+   * @private
+   */
+  _patch(data) {
+    /**
+     * The id of the guild.
+     * @type {Snowflake}
+     */
+    this.id = data.id;
+
+    /**
+     * The name of the guild.
+     * @type {string}
+     */
+    this.name = data.name;
+
+    /**
+     * The invite of the guild.
+     * @type {?string}
+     */
+    this.instantInvite = data.instant_invite;
+
+    /**
+     * The list of channels in the guild.
+     * @type {Collection<Snowflake, WidgetChannel>}
+     */
+    this.channels = new Collection();
+    for (const channel of data.channels) {
+      this.channels.set(channel.id, channel);
+    }
+
+    /**
+     * The list of members in the guild.
+     * These strings are just arbitrary numbers, they aren't Snowflakes.
+     * @type {Collection<string, WidgetMember>}
+     */
+    this.members = new Collection();
+    for (const member of data.members) {
+      this.members.set(member.id, member);
```

Comment:
I just noticed this is raw data from Discord which is exposed to the user, and as such, users would require to use `snake_case` for some properties (`self_deaf`, `self_mute`...) when reading them.

Shouldn't this have its own structure?